packages/components/popconfirm/src/popconfirm.vue

```diff
@@ -90,11 +94,17 @@ const confirm = (e: MouseEvent) => {
   emit('confirm', e)
   hidePopper()
 }
-const cancel = (e: MouseEvent) => {
+const cancel = (e: MouseEvent | KeyboardEvent) => {
   emit('cancel', e)
   hidePopper()
 }
 
+const onCloseRequested = (event: KeyboardEvent) => {
+  if (props.closeOnPressEscape && event.code === EVENT_CODE.esc) {
```

Comment:
```suggestion
  if (props.closeOnPressEscape) {
```
https://github.com/element-plus/element-plus/blob/b04b827a1fba0175e316a377b2d6775b296aba28/packages/hooks/use-escape-keydown/index.ts#L8

The judgment has been made inside the hook.